### PR TITLE
build akka-cluster-typed on scala 3

### PR DESCRIPTION
**Pull request by raboof**
Refs #30243

### CODE REVIEW

File: akka-cluster-typed/src/main/scala-3/akka/cluster/typed/internal/receptionist/ClusterReceptionistProtocol.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[receptionist] object ClusterReceptionistProtocol {
+  type Aux[P] = AbstractServiceKey { type Protocol = P }
+
+  type SubscriptionsKV[K <: Aux[_]] = K match {
+    case Aux[t] => ActorRef[ReceptionistMessages.Listing[t]]
```
Comment (octonato) [inline akka-cluster-typed/src/main/scala-3/akka/cluster/typed/internal/receptionist/ClusterReceptionistProtocol.scala:19]: Please, remind me again what is the trick with `t` (lowercase).

**Review by octonato — APPROVED** [review]

File: akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/BasicClusterExampleTest.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public void clusterApiExample() {
 
       // #prepare
       PrepareForFullClusterShutdown msg =
-          PrepareForFullClusterShutdown.prepareForFullClusterShutdown();
+          PrepareForFullClusterShutdown$.MODULE$.prepareForFullClusterShutdown();
```
Comment (johanandren) [inline akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/BasicClusterExampleTest.java:72]: Ugh, that's no nice Java API.

File: akka-cluster-typed/src/main/scala-3/akka/cluster/typed/internal/receptionist/ClusterReceptionistProtocol.scala (hunk 1 above)
Comment (raboof) [reply akka-cluster-typed/src/main/scala-3/akka/cluster/typed/internal/receptionist/ClusterReceptionistProtocol.scala:19]: in this case it's a variable in the type-level patternmatch, https://blog.rockthejvm.com/scala-3-match-types/

File: akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/BasicClusterExampleTest.java (hunk 2 above)
Comment (raboof) [reply akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/BasicClusterExampleTest.java:72]: yeah, we might want to lobby for those static forwarders to be generated again in the scala compiler - I haven't looked into why they're not yet.

**Review by johanandren — APPROVED** [review]

**Pull request merged**